frontend/js/src/user/components/follow/UserSocialNetwork.tsx

```diff
@@ -69,6 +80,82 @@ export default class UserSocialNetwork extends React.Component<
     }
   };
 
+  getSimilarity = async () => {
+    const { user } = this.props;
+    const { APIService, currentUser } = this.context;
+    if (currentUser?.name === user?.name) {
+      return;
+    }
+    const { getSimilarityBetweenUsers } = APIService;
+    try {
+      const response = await getSimilarityBetweenUsers(
+        user.name,
+        currentUser.name
+      );
+      const { payload } = response;
+      const similarityScore = payload.similarity;
+      this.setState({ similarityScore });
+    } catch (err) {
+      if (err.toString() === "Error: Similar-to user not found") {
+        // cannot get similarity if not in the list, so just return 0
+        this.setState({ similarityScore: 0 });
+      } else {
+        toast.error(
+          <ToastMsg
+            title="Error while fetching similarity"
+            message={err.toString()}
+          />,
+          { toastId: "fetch-similarity-error" }
+        );
+      }
+    }
+  };
+
+  getSimilarArtists = async () => {
+    const { user } = this.props;
+    const { APIService, currentUser } = this.context;
+    if (currentUser?.name === user?.name) {
+      return;
+    }
+    const { getUserEntity } = APIService;
+    try {
+      let response = await getUserEntity(
+        user.name,
+        "artist",
+        "all_time",
+        0,
+        100
+      );
+      const userArtists = (response as UserArtistsResponse).payload.artists;
+      response = await getUserEntity(
+        currentUser.name,
```

Comment:
This currently fetches all time global stats when a user is not logged in, so you end up with whatever intersection there is between the user whose page we're looking at and the global stats.
We should exit early if there is no currentUser, as commented above.